sds/src/secondary_validation/bulgarian_egn_checksum.rs

```diff
@@ -0,0 +1,69 @@
+use crate::secondary_validation::{get_next_digit, Validator};
+
+pub struct BulgarianEGNChecksum;
+
+const BULGARIAN_EGN_MULTIPLIERS: [u32; 9] = [2, 4, 8, 5, 10, 9, 7, 3, 6];
+
+impl Validator for BulgarianEGNChecksum {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        if regex_match.len() != 10 {
+            return false;
+        }
+
+        // Convert the string to a vector of digits
+        let digits: Vec<u32> = regex_match.chars().filter_map(|c| c.to_digit(10)).collect();
+        if digits.len() != 10 { // any non-digit character would mark this invalid
+            return false;
+        }
+
+        // calculate sum(chars[i] * BULGARIAN_EGN_MULTIPLIERS[i]) for i in [0,9)
+        let sum: u32 = digits.iter().take(9)
+            .zip(BULGARIAN_EGN_MULTIPLIERS.iter())
+            .map(|(d, w)| d * w)
+            .sum();
+
+        // The check digit is the last digit of the sum modulo 11, with a special case for 10
+        let mut check_digit = sum % 11;
+        if check_digit == 10 {
+            check_digit = 0;
+        }
```

Comment:
can also do a `let mut check_digit = sum % 11 % 10`, but the current way is more explicit and verbose thus opted for this way.